duecredit/io.py

```diff
@@ -340,9 +358,18 @@ def format_bibtex(bibtex_entry: BibTeX, style: str = "harvard1") -> str:
             # return warnings back
             warnings.filters = old_filters
         bib_style = cp.CitationStylesStyle(style, validate=False)
-        # TODO: specify which tags of formatter we want
+        formatters = {
+            "plain": cp.formatter.plain,
+            "html": cp.formatter.html,
+            "rst": cp.formatter.rst,
+        }
```

Comment:
futureproof and make it more flexible!

```suggestion
        formatters = {
            f: (fmt:=getattr(cp.formatter, f))
            for f in dir(cp.formatter)
            if (not f.startswith('_')) and hasattr(fmt, 'preformat')
        }
```

although we could even just remove that check on preformat and use all public:

```suggestion
        formatters = {
            f: getattr(cp.formatter, f)
            for f in dir(cp.formatter)
            if not f.startswith('_')
        }
```